listeners.py

```diff
@@ -197,13 +199,16 @@ def process_result(self, obj):
         # called only once on task run initiated
         if isinstance(obj, WorkerCurrentTask):
             task_name = self.get_task_name_conf(obj.task_name, obj.task_param)
+            self.tasks_timings[task_name] = time.time()
             self.worker_current_task[obj.worker_id] = obj
             self.rss_results[task_name] = get_proc_stat_rss(
                     self.worker_id_to_pid[obj.worker_id])
 
         # called only once on task run finished
         if isinstance(obj, WorkerTaskResult):
             task_name = self.get_task_name_conf(obj.task_id[0], obj.task_id[1])
+            self.tasks_timings[task_name] = round(time.time() -
+                                                  self.tasks_timings[task_name], 2)
```

Comment:
I would add test timing into `WorkerTaskResult`, because of several reasons:

1. Accuracy. We process many different results here and the moment of this `time.time()` is not strictly after the end of a test.
2. Don't sum timings re-runs of flaky tests.
3. Reliability. If we'll add some heavy work into the main process (intentionally or not), timings will be affected if we'll collect them from the main process. While tracking from the worker process will achieve the best possible precision.

The worst thing here is that even if everything works nice during testing of the patch, we can lose accuracy later in a specific environment or after some test-run changes and **we will not know about this**. That's why I insist on the approach that will ensure best possible accuracy. It is not a thing that either work or report an error.

Reply:
Sure, moved get finish time to the "WorkerTaskResult", also start time to the "WorkerCurrentTask".